scheduler/src/cook/mesos/reason.clj

```diff
@@ -13,78 +13,34 @@
 ;; See the License for the specific language governing permissions and
 ;; limitations under the License.
 ;;
-(ns cook.mesos.reason)
+(ns cook.mesos.reason
+  (:require [datomic.api :as d]
+            [clojure.tools.logging :as log]))
 
-;; Note: Commented reasons are specified but not implemented
+(defn reason-code->reason-entity
+  [db reason-code]
+  (d/entity db [:reason/code reason-code]))
 
-;;(def reason-normal 1000)
-;;(def reason-aborted-by-user 1001)
-(def reason-preempted-by-rebalancer 1002)
-(def reason-mesos-container-preempted 1003)
+(defn reason-code->reason-string
+  [db reason-code]
+  (:reason/string (reason-code->reason-entity db reason-code)))
 
-(def reason-mesos-container-limitation 2000)
-(def reason-mesos-container-limitation-disk 2001)
-(def reason-mesos-container-limitation-memory 2002)
-(def reason-max-runtime 2003)
+(defn mesos-reason->cook-reason-entity-id
+  [db mesos-reason]
+  (if-let [reason-entity-id (:db/id (d/entity db [:reason/mesos-reason mesos-reason]))]
+    reason-entity-id
+    (do
+      (log/warn "Unknown mesos reason: " mesos-reason)
+      (:db/id (d/entity db [:reason/name :mesos-unknown])))))
```

Comment:
So the point of this is to store mesos' original reason for the job being cancelled, as well as Cook's understanding of that reason (if any), and to do that regardless of whether Cook is aware of all of the possible Mesos reasons - right?